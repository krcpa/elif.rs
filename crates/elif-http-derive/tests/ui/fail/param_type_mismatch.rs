use elif_http_derive::{get, param};

#[get("/users/{id}")]
#[param(id: string)]  // Mismatch: route has {id} but param declares string
fn get_user(id: u32) -> String {  // Function expects u32, param declares string
    format!("User {}", id)
}