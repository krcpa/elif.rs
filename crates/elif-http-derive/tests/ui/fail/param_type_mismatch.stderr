error: Parameter type mismatch: Parameter 'id' has type 'u32' but expected one of: String, &str, str. Hint: Change the function parameter to match #[param(id: string)] or change the #[param] declaration to match the function parameter.
 --> tests/ui/fail/param_type_mismatch.rs:5:1
  |
5 | fn get_user(id: u32) -> String {  // Function expects u32, param declares string
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/ui/fail/param_type_mismatch.rs:7:2
  |
7 | }
  |  ^ consider adding a `main` function to `$DIR/tests/ui/fail/param_type_mismatch.rs`
